### DISCUSSION TITLE
Getting an error when trying to mixin to the ItemEntity class please help me out.

### DISCUSSION THREAD

**Discussion by TechPro424** [FabricMC/fabric-api#2811] [Mod Dev Support] [answered]
Getting an error when trying to mixin to the ItemEntity class please help me out.
https://gist.github.com/TechPro424/3564c0629963e903a7e9931fb6e61b52

**Comment by warjort** [comment, marked as answer]
You mixing into the ItemEntity class, but then specifying the method is on the Entity class
````
Lnet/minecraft/entity/Entity;onPlayerCollision
````
Change the Entity class to ItemEntity class name.
Or for simple cases, you can just specify the method name and it will match against the arguments of your mixin method.

You are also trying to use LocalCapture to get access to variables that won't exist at "HEAD" - the start of the method?
If you want access to the ItemStack you need to use Shadow fields/methods
https://fabricmc.net/wiki/tutorial:mixin_tips

**Reply by TechPro424** [reply]
Thanks, that worked! 
I ended up specifying the method is on the Entity class because that's what it gave me when I ran `!ym onPlayerCollision 1.19.3` in the Fabric Discord and never realized it was pointing to the Entity class
